File: services/tradelog/tradelog-api/src/main/java/com/example/tradelog/api/repository/DividendModelRowMapper.java
package com.example.tradelog.api.repository;

import com.example.tradelog.api.core.model.DividendJournalModel;
import com.example.tradelog.api.core.model.TransactionModel;
import org.springframework.jdbc.core.RowMapper;

import java.sql.ResultSet;
import java.sql.SQLException;

public class DividendModelRowMapper implements RowMapper<DividendJournalModel> {
    @Override
    public DividendJournalModel mapRow(ResultSet rs, int rowNum) throws SQLException {
        TransactionModel transactionModel = new TransactionModelRowMapper(rs).invoke();

        return new DividendJournalModel(
                transactionModel,
                rs.getDouble("dividend"),
                rs.getInt("quantity"));
    }
}

File: services/tradelog/tradelog-api/src/main/kotlin/com/example/tradelog/api/db/converter/DividendModelRowMapper.kt
package com.example.tradelog.api.db.converter

import com.example.tradelog.api.core.model.DividendJournalModel
import org.springframework.jdbc.core.RowMapper
import java.sql.ResultSet

class DividendModelRowMapper: RowMapper<DividendJournalModel> {

    override fun mapRow(rs: ResultSet, rowNum: Int): DividendJournalModel? {
        val transactionModel = TransactionModelRowMapper(rs).invoke()

        return DividendJournalModel(transactionDetails = transactionModel,
                dividend = rs.getDouble("dividend"),
                quantity = rs.getInt("quantity")
        )
    }
}